substrate/frame/revive/fixtures/build.rs

```diff
@@ -85,19 +85,30 @@ fn create_cargo_toml<'a>(
 	let mut cargo_toml: toml::Value = toml::from_str(include_str!("./build/_Cargo.toml"))?;
 	let uapi_dep = cargo_toml["dependencies"]["uapi"].as_table_mut().unwrap();
 
-	let metadata = MetadataCommand::new()
-		.manifest_path(fixtures_dir.join("Cargo.toml"))
-		.exec()
-		.expect("Failed to fetch cargo metadata");
+	let manifest_path = fixtures_dir.join("Cargo.toml");
+	let metadata = MetadataCommand::new().manifest_path(&manifest_path).exec().unwrap();
+	let dependency_graph = metadata.resolve.unwrap();
 
-	let mut uapi_pkgs: Vec<_> = metadata
+	// Resolve the pallet-revive-fixtures package id
+	let fixtures_pkg_id = metadata
 		.packages
 		.iter()
-		.filter(|pkg| pkg.name == "pallet-revive-uapi")
-		.collect();
+		.find(|pkg| pkg.manifest_path.as_std_path() == manifest_path)
```

Comment:
We are still just picking the first package we find here. Why is the manifest path unique but the name is not?

Reply:
They look like that 
```
      "manifest_path": "/home/pg/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pallet-contracts-uapi-12.0.0/Cargo.toml",
      "manifest_path": "/home/pg/github/polkadot-sdk/substrate/frame/contracts/uapi/Cargo.toml",
      "manifest_path": "/home/pg/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pallet-revive-uapi-0.1.1/Cargo.toml",
```
so we can pick the right one knowing the path to the manifest

Reply:
there might be a better approach but that seem to work ...